### PR TITLE
Add generate code tools.

### PR DESCRIPTION
**Pull request by TommyLike**
**What this PR does / why we need it**:
Add generate code tools and make command

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
/lgtm

please fix the verify

File: Makefile [hunk 1]
```diff
@@ -71,6 +71,9 @@ run-test:
 e2e: kube-batch
 	hack/run-e2e.sh
 
+generate-code:
```
Comment (hzxuzhonghu) [inline Makefile:74]: I would suggest `generate-code: generate-deepcopy`, so developers can call only `make generate-code `

Comment (TommyLike) [reply Makefile:74]: The function of ``generate-deepcopy`` is already been included into ``generate-code``

Comment (hzxuzhonghu) [reply Makefile:74]: I noticed that, can i say the `generate-deepcopy` is a redundant or a legacy stuff, which can be removed?

**Pull request merged**